personal_budget_main.py

```diff
@@ -0,0 +1,214 @@
+                        
+                        txn_key = "".join(random.choices(string.ascii_uppercase + string.digits, k=6)) # Unique key to identify individual transaction.
+                        transactions_all[txn_key] = {   "txn_date" : txn_date.strftime("%d-%b-%Y"),
+                                                        "txn_detail" : txn_detail,
+                                                        "debit" : debit,
+                                                        "credit" : credit
+                                                    }
+                        
+                        # Consolidating transactions month-wise.
+                        month_key = txn_date.strftime("%m-%Y")
+                        if month_key not in monthly_summary:
+                            monthly_summary[month_key] = {  "month_name" : txn_date.strftime("%B, %Y"),
+                                                            "txn_keys" : [txn_key],
+                                                            "debit" : debit,
+                                                            "credit" : credit,
+                                                            "surplus_deficit" : credit - debit
+                                                        }
+                        else:
+                            monthly_summary[month_key]["txn_keys"].append(txn_key)
+                            monthly_summary[month_key]["debit"] += debit
+                            monthly_summary[month_key]["credit"] += credit
+                            monthly_summary[month_key]["surplus_deficit"] += credit - debit
+        
+        if not excel_file_exist:
+            sys.exit("No Excel file in source directory")
+    
+    except Exception as e:
+        print(f"Error: {e}")
+        sys.exit(1)
+    
+    return transactions_all, monthly_summary
+
+
+def generate_report(transactions_all, monthly_summary, report_meta, sort_code):
+    """
+    Generates output data
+    """
+    final_report = []
+    
+    # Generates month-wise summary.
+    if report_meta.get("type", "M") == "M":
+        table = [[month_detail["month_name"], month_detail["debit"], \
+           month_detail["credit"], month_detail["surplus_deficit"]] \
+           for month_detail in monthly_summary.values()]
+        if not sort_code.get("sort_req", False):
+            final_report = table
+        else:
+            sort_by_column = 1 if sort_code.get("sort_by", "D") == "D" else 2 if sort_code.get("sort_by", "D") == "C" else 3
+            descend = True if sort_code.get("order", "D") == "D" else False
+            i = 0
+            while i < len(table):
+                if table[i][sort_by_column] == 0:
+                    table.pop(i)
+                else:
+                    i += 1
+            final_report = sorted(table, key = lambda x : x[sort_by_column], reverse = descend)
```

Comment:
reverse = descend, key = lambda x

There should not be spaces next to the = operator for parameters of a function